module/VuFind/tests/unit-tests/src/VuFindTest/Http/ServerUrlHelperTest.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * ServerUrlHelper Test Class
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Http;
+
+use Closure;
+use Laminas\View\Helper\ServerUrl;
+use VuFind\Http\ServerUrlHelper;
+
+/**
+ * ServerUrlHelper Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class ServerUrlHelperTest extends \PHPUnit\Framework\TestCase
+{
+    use \VuFindTest\Feature\ViewTrait;
+
+    /**
+     * Specific $_SERVER fields to backup before tests and restore after
+     *
+     * @var array
+     */
+    protected array $serverBackupFields = ['HTTPS', 'HTTP_HOST', 'REQUEST_URI'];
+
+    /**
+     * Data provider to return in getBaseUrl tests.
+     *
+     * @return \Iterator<(int | string), array<mixed>>
```

Comment:
We can be more specific than the auto-generated hint:

```suggestion
     * @return \Iterator<string, array>
```